import-automation/executor/app/service/github_api.py

```diff
@@ -0,0 +1,374 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+GitHub API client for querying information about a repository.
+"""
+
+import os
+import tarfile
+import tempfile
+import http
+from typing import Dict, Set, List
+
+import requests
+
+from app import utils
+
+_GITHUB_API_HOST = 'https://api.github.com'
+
+_GITHUB_COMMIT_API = (
+    _GITHUB_API_HOST +
+    '/repos/{owner_username}/{repo_name}/commits/{commit_sha}')
+
+_GITHUB_CONTENT_API = (
+    _GITHUB_API_HOST +
+    '/repos/{owner_username}/{repo_name}/contents/{dir_path}?ref={commit_sha}')
+
+_GITHUB_USER_API = _GITHUB_API_HOST + '/users/{name}'
+
+_GITHUB_DOWNLOAD_API = (
+    _GITHUB_API_HOST +
+    '/repos/{owner_username}/{repo_name}/tarball/{commit_sha}')
+
+
+class GitHubRepoAPI:
+    """GitHub API client for querying information about a repository.
+
+    The methods check for the response status code and throw
+    requests.exceptions.HTTPError if the code is larger than or equal to 400.
+
+    Attributes:
+        owner: Username of the owner of the repository as a string.
+        repo: Name of the repository as a string.
+        auth: Tuple consisting of the username of the account to authenticate
+            with GitHub and the access token.
+    """
+
+    def __init__(self,
+                 repo_owner_username: str,
+                 repo_name: str,
+                 auth_username: str = '',
+                 auth_access_token: str = ''):
+        """Constructs a GitHubRepoAPI.
+
+        Args:
+            repo_owner_username: See owner in Attributes.
+            repo_name: See repo in Attributes.
+            auth_username: The username of the account to authenticate
+                with GitHub, as a string.
+            auth_access_token: The corresponding access token as a string.
+        """
+        self.owner = repo_owner_username
+        self.repo = repo_name
+        self.auth = (auth_username, auth_access_token)
+
+    def query_commit(self, commit_sha: str) -> Dict:
+        """Queries the information about a commit to the repository.
+
+        Args:
+            commit_sha: ID of the commit as a string.
+
+        Returns:
+            Information about the commit as a dict.
+        """
+        commit_query = self._build_commit_query(commit_sha)
+        response = requests.get(commit_query, auth=self.auth)
```

Comment:
If the commit sha is valid, yes. Otherwise, it throws an exception.

```
curl -I https://api.github.com/repos/datacommonsorg/data/commits/123
-> status: 422 Unprocessable Entity

```